### PR TITLE
Continuous Applications: Delete CourseChoices SiteSelection

### PR DESCRIPTION
**Pull request by inulty-dfe**
## Context

Continuous Applications changed how Candidates pick courses.

During the development of Continuous Applications, a new separate path and set of files were produced to allow Candidates choose a Course and to facilitate the migration from the old Course Choice flow to the Continuous Applications.

This change is part of a series of PRs to delete the old CourseChoice flow and namespace.

## Changes proposed in this pull request

The `CourseChoiceReviewComponent` is still being used when a candidate views an old applciation from a previous cycle.
This component used to allow for editing, though now there is no scope for editing old applications so the editing functionality has been removed.

Delete `CourseChoices::SiteSelectionController`, views, routes and links.

Delete all specs that fail as a consequence of this change.

## Guidance to review

Confirm that removing editable from `CourseChoiceReviewComponent` is ok. 
## Link to Trello card

[Trello Ticket](https://trello.com/c/wi1cHFXH/1086-catd-coursechoices-siteselection)

### CODE REVIEW

**Comment by inulty-dfe** [comment]
[CATD: CourseChoices => SiteSelection](https://trello.com/c/wi1cHFXH/1086-catd-coursechoices-siteselection)

File: app/components/candidate_interface/course_choices_review_component.html.erb [hunk 1]
```diff
@@ -1,46 +1,6 @@
...
-          title: 'You cannot apply to this course as the chosen study mode is full',
-          alternatives: [
-            (govuk_link_to('pick a new location', site_change_path(application_choice)) if site_change_path(application_choice)),
-          ],
-        )) %>
-      <% end %>
-    <% end %>
-
-    <%= render(SummaryCardComponent.new(rows: course_choice_rows(application_choice), editable: @editable)) do %>
+    <%= render(SummaryCardComponent.new(rows: course_choice_rows(application_choice), editable: false)) do %>
```
Comment (vassyz) [inline app/components/candidate_interface/course_choices_review_component.html.erb:3]: Is it not `false` by default?

File: app/components/candidate_interface/course_choices_review_component.rb [hunk 2]
```diff
@@ -55,30 +53,6 @@ def any_withdrawable?
       end
     end
 
-    def show_missing_banner?
-      @show_incomplete && !@application_form.course_choices_completed && @editable
-    end
-
-    def course_change_path(application_choice)
-      if multiple_courses?(application_choice) || application_choice.course_full?
```
Comment (vassyz) [inline app/components/candidate_interface/course_choices_review_component.rb:63]: Shouldn't the `multiple_courses?` and `multiple_sites?` methods be deleted as well?

**Review by vassyz — APPROVED** [review]
Looks good, left some comments.

File: app/controllers/candidate_interface/course_choices/site_selection_controller.rb [hunk 3]
```diff
@@ -1,70 +0,0 @@
...
-            id_of_course_choice_to_replace: params[:course_choice_id],
-            return_to: params[:return_to],
-          )
-          .call
-      end
-
-    private
-
-      def available_sites
-        PickSiteForm.available_sites(params.fetch(:course_id), params.fetch(:study_mode))
```
Comment (vassyz) [inline app/controllers/candidate_interface/course_choices/site_selection_controller.rb:65]: I think you can also delete `self.available_sites` from `class PickSiteForm`.

File: app/components/candidate_interface/course_choices_review_component.html.erb (hunk 1 above)
Comment (inulty-dfe) [reply app/components/candidate_interface/course_choices_review_component.html.erb:3]: It's true by default

https://github.com/DFE-Digital/apply-for-teacher-training/tree/main/app/components/utility/summary_card_component.rb#L2

**Pull request merged**